gnovm/pkg/gnolang/preprocess.go

```diff
@@ -2540,6 +2354,219 @@ func preprocess1(store Store, ctx BlockNode, n Node) Node {
 	return nn
 }
 
+// defineOrDecl merges the code logic from op define (:=) and declare (var/const).
+func defineOrDecl(
+	store Store,
+	bn BlockNode,
+	isConst bool,
+	nameExprs []NameExpr,
+	typeExpr Expr,
+	valueExprs []Expr,
+) {
+	numNames := len(nameExprs)
+	numVals := len(valueExprs)
+
+	if numVals > 1 && numNames != numVals {
+		panic(fmt.Sprintf("assignment mismatch: %d variable(s) but %d value(s)", numNames, numVals))
+	}
+
+	if numNames < 1 {
```

Comment:
Normally I think that cannot happen in this context, but this check was  just to ensure that the function does all the needed checks.

But you are right, this is only used in the context where we always have at least 1 Lhs expr (if not it will raise a syntax error earlier) so this is unneeded check